src/contracts/AbstractARM.sol

```diff
@@ -0,0 +1,652 @@
+        }
+
+        // Need to remove the liquidity assets that have been reserved for the withdrawal queue
+        return assets + claimedMem - queuedMem;
+    }
+
+    /// @dev Hook for calculating the amount of assets in an external withdrawal queue like Lido or OETH
+    /// This is not the ARM's withdrawal queue
+    function _externalWithdrawQueue() internal view virtual returns (uint256 assets);
+
+    /// @notice Calculates the amount of shares for a given amount of liquidity assets
+    function convertToShares(uint256 assets) public view returns (uint256 shares) {
+        uint256 totalAssetsMem = totalAssets();
+        shares = (totalAssetsMem == 0) ? assets : (assets * totalSupply()) / totalAssetsMem;
+    }
+
+    /// @notice Calculates the amount of liquidity assets for a given amount of shares
+    function convertToAssets(uint256 shares) public view returns (uint256 assets) {
+        assets = (shares * totalAssets()) / totalSupply();
+    }
+
+    /// @notice Set the Liquidity Provider Controller contract address.
+    /// Set to a zero address to disable the controller.
+    function setLiquidityProviderController(address _liquidityProviderController) external onlyOwner {
+        liquidityProviderController = _liquidityProviderController;
+
+        emit LiquidityProviderControllerUpdated(_liquidityProviderController);
+    }
+
+    ////////////////////////////////////////////////////
+    ///         Performance Fee Functions
+    ////////////////////////////////////////////////////
+
+    /// @notice Owner sets the performance fee on increased assets
+    /// @param _fee The performance fee measured in basis points (1/100th of a percent)
+    /// 10,000 = 100% performance fee
+    /// 500 = 5% performance fee
+    function setFee(uint256 _fee) external onlyOwner {
+        _setFee(_fee);
+    }
+
+    /// @notice Owner sets the account/contract that receives the performance fee
+    function setFeeCollector(address _feeCollector) external onlyOwner {
+        _setFeeCollector(_feeCollector);
+    }
+
+    function _setFee(uint256 _fee) internal {
+        require(_fee <= FEE_SCALE, "ARM: fee too high");
+
+        // Collect any performance fees up to this point using the old fee
+        collectFees();
+
+        fee = SafeCast.toUint16(_fee);
+
+        emit FeeUpdated(_fee);
+    }
+
+    function _setFeeCollector(address _feeCollector) internal {
+        require(_feeCollector != address(0), "ARM: invalid fee collector");
+
+        feeCollector = _feeCollector;
+
+        emit FeeCollectorUpdated(_feeCollector);
+    }
+
+    /// @notice Transfer accrued performance fees to the fee collector
+    /// This requires enough liquidity assets in the ARM to cover the accrued fees.
+    function collectFees() public returns (uint256 fees) {
+        uint256 newAvailableAssets;
+        // Accrue any performance fees up to this point
+        (fees, newAvailableAssets) = _feesAccrued();
+
+        if (fee == 0) return fees;
```

Comment:
⚪️ Personal preference - I think `return 0` is clearer here. Doesn't require the mental step of thinking through the implications of when this is called. Also it looks like our standard, "we are going to do nothing and exit early".

Reply:
I agree. I've changed